src/test/java/nl/tudelft/ewi/devhub/server/backend/warnings/LargeCommitWarningGeneratorTest.java

```diff
@@ -0,0 +1,137 @@
+package nl.tudelft.ewi.devhub.server.backend.warnings;
+
+import com.google.common.collect.Sets;
+import nl.tudelft.ewi.devhub.server.database.entities.Commit;
+import nl.tudelft.ewi.devhub.server.database.entities.RepositoryEntity;
+import nl.tudelft.ewi.devhub.server.database.entities.warnings.LargeCommitWarning;
+import nl.tudelft.ewi.devhub.server.web.models.GitPush;
+import nl.tudelft.ewi.git.models.AbstractDiffModel;
+import nl.tudelft.ewi.git.models.DiffModel;
+import nl.tudelft.ewi.git.web.api.CommitApi;
+import nl.tudelft.ewi.git.web.api.RepositoriesApi;
+import nl.tudelft.ewi.git.web.api.RepositoryApi;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LargeCommitWarningGeneratorTest {
+
+    private LargeCommitWarningGenerator generator;
+    private Set<LargeCommitWarning> testEquals;
+
+    private static final String REPOSITORY_NAME = "John Cena";
+    private static final String COMMIT_ID = "1";
+
+    @Mock private RepositoriesApi repositoriesApi;
+    @Mock private Commit commit;
```

Comment:
Also, data objects tend to be immutable (this is not completely the case for database entities which are being observed for changes though). Because they are immutable, it is also unlikely that you would want to verify an invocation on the object, the second purpose of mocking :)